src/cascadia/TerminalCore/TerminalDispatch.cpp

```diff
@@ -399,6 +402,63 @@ bool TerminalDispatch::EndHyperlink() noexcept
     return _terminalApi.EndHyperlink();
 }
 
+// Method Description:
+// - Performs a ConEmu action
+// - Currently, the only action we support is setting the taskbar state/progress
+// Arguments:
+// - string: contains the parameters that define which action we do
+// Return Value:
+// - true
+bool TerminalDispatch::DoConEmuAction(const std::wstring_view string) noexcept
+{
+    unsigned int state = 0;
+    unsigned int progress = 0;
+
+    const auto parts = Utils::SplitString(string, L';');
+
+    unsigned int subParam = 0;
+    const auto subParamSuccess = Utils::StringToUint(til::at(parts, 0), subParam);
+
+    // For now, the only ConEmu action we support is setting the taskbar state/progress,
+    // which has a sub param value of 4
+    if (parts.size() < 1 || !subParamSuccess || subParam != 4)
+    {
+        return false;
+    }
+
+    if (parts.size() == 2)
+    {
+        // A state parameter is defined, parse it out
+        const auto stateSuccess = Utils::StringToUint(til::at(parts, 1), state);
+        if (!stateSuccess)
+        {
+            return false;
+        }
+    }
+    else if (parts.size() > 2)
+    {
+        // Both state and progress are defined, parse them out
+        const auto stateSuccess = Utils::StringToUint(til::at(parts, 1), state);
+        const auto progressSuccess = Utils::StringToUint(til::at(parts, 2), progress);
+        if (!stateSuccess || !progressSuccess)
+        {
+            return false;
+        }
+    }

```

Comment:
is it possible to fold these together?

```
If it's >= 1, parse state
If it failed, return
   If it's >= 2, parse progress
   If it failed, return
```

Reply:
This actually made me realize there's an issue here - I only checked for the size of `parts` _after_ already calling `til::at(parts, 0)`, which means a crash occurs if we send in just ``e]9;`e` - will fix that now